File: LinkedList.java
public class LinkedList<T> {
    class Node {
        T val;
        Node next;

        Node(T data) {
            val = data;
            next = null;
        }
    }

    Node head;

    LinkedList() {
        head = null;
    }

    public void InsetAtBegging(T data) {
        Node newNode = new Node(data);

        if (head == null) {
            head = newNode;
        } else {
            newNode.next = head;
            head = newNode;
        }
    }

    public void display() {// linked list traversal
        Node temp = head;
        System.out.print("datas are ");
        while (temp != null) {
            System.out.print(temp.val + " ");
            temp = temp.next;
        }
        System.out.println();
    }

    public void InsertAtLast(T data) {

        if (head == null) {
            throw new IndexOutOfBoundsException();
        }
        Node newNode = new Node(data);
        Node temp = head;
        Node prev = head;
        while (temp != null) {
            prev = temp;
            temp = temp.next;
        }
        prev.next = newNode;
    }

    public void insertAtPos(T data, int pos) {
        if (pos == 0) {
            InsetAtBegging(data);
            return;
        }

        Node newNode = new Node(data);
        Node temp = head;
        for (int i = 0; i < pos - 1; i++) {
            temp = temp.next;
            if (temp == null) {
                System.out.println("Invaild");
                return;
            }
        }
        newNode.next = temp.next;
        temp.next = newNode;
    }

    public void delAtBeg() {
        head = head.next;
    }

    public void delAtPos(int pos) {
        Node temp = head;
        Node prev = head;
        if (pos == 0) {
            delAtBeg();
            return;
        }
        for (int i = 0; i < pos; i++) {
            prev = temp;
            temp = temp.next;
        }

        prev.next = temp.next;

    }

    public void reverse() {
        Node prev = null;
        Node current = head;
        Node next = head.next;
        while (current != null) {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }
        head = prev;
    }
}

File: Linkedlist.c
#include<stdio.h>
#include<stdlib.h>

struct Node
{
    int val;
    struct Node *next;
};

struct Node *head=NULL;

void insertAtbeg(int val){
    struct Node *ptr;
    ptr=(struct Node *)malloc(sizeof(struct Node ));
    if(ptr==NULL){
        printf("Over flow");
        return;
    }
    else{
        ptr->val=val;
        ptr->next=head;
        head=ptr;
    }
}
void insertAtlast(int val){
    if(head==NULL){
        head->val=val;
        head->next=NULL;
        return;
    }
    struct Node *temp=head;
    struct Node *ptr=(struct Node *)malloc(sizeof(struct Node));
    while(temp->next!=NULL){
        temp=temp->next;
    }
    ptr->val=val;
    temp->next=ptr;
    ptr->next=NULL;
}

void inserAtpos(int val,int pos){
    if(pos==0){
        insertAtbeg( val);
        return;
    }
    struct Node *temp=head;
    struct Node *pre=NULL;
    struct Node *ptr=(struct Node *)malloc(sizeof(struct Node));
    for(int i=0;i<pos-1;i++){
           pre=temp;
           temp=temp->next;
           if(temp==NULL){
            printf("invaild index \n");
            return;
           }
    }
    ptr->val=val;
    ptr->next=temp;
    pre->next=ptr;

}

void delAtbeg(){
    if(head==NULL){
        printf("under flow\n");
        return;
    }
    head=head->next;
}

void delAtlast(){
    struct Node *temp=head;
    struct Node *ptr=NULL;

    while(temp->next!=NULL){
        ptr=temp;
        temp=temp->next;
    }
    ptr->next=NULL;
}

void delAtpos(int pos){
    struct Node *temp=head;
    struct Node *pre=NULL;
    if(pos==0){
        delAtbeg();
        return;
    }
    for(int i=0;i<pos-1;i++){
        pre=temp;
        temp=temp->next;
        if(temp==NULL){
            printf("Invaild position\n");
            return;
        }
    }
    pre->next=temp->next;
}

void display(){
    struct Node *temp;
    temp=head;
    while(temp!=NULL){
        printf("%d ",temp->val);
        temp=temp->next;
    }
    printf("\n");
}


int main(){
    insertAtbeg(5);
    insertAtbeg(10);
    display();
    insertAtlast(1);
    insertAtlast(2);
    display();
    inserAtpos(6,3);
    display();
    inserAtpos(0,0);
    display();
    inserAtpos(10,10);
    delAtbeg();
    display();
    delAtlast();
    display();
    delAtpos(3);
    display();
    delAtpos(10);
    

    
}